lib/scoped/find.js

```diff
@@ -1,15 +1,20 @@
 module.exports = find
 
+var isNotFoundError = require('../utils/is-not-found-error')
 var typeFilter = require('../utils/type-filter')
 
 function find (type, api, objectsOrIds) {
   return api.find(objectsOrIds)
 
   .then(function (storedObjects) {
     if (Array.isArray(storedObjects)) {
-      var typedObjects = storedObjects.filter(typeFilter.bind(null, type))
+      var typedObjects = storedObjects.map(function (storedObject) {
+        return (storedObject instanceof Error || typeFilter(type, storedObject))
+          ? storedObject
+          : handleIncorrectTypeError(type, storedObject)
+      })
 
-      if (typedObjects.length === 0) {
+      if (allObjectsAreNotFound(typedObjects)) {
         throw new Error('Items not found')
```

Comment:
I’d argue that we shouldn’t throw an error here. If I call `store.find()` with an array, it should always resolve with an array, even none of the documents could be found. I’d vote for predictability here